presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java

```diff
@@ -105,6 +110,21 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         response.sendError(SC_UNAUTHORIZED, Joiner.on(" | ").join(messages));
     }
 
+    private boolean doesRequestSupportAuthentication(HttpServletRequest request)
+    {
+        if (authenticators.isEmpty()) {
+            return false;
+        }
+        if (request.isSecure()) {
+            return true;
+        }
+        if (securityConfig.getEnableForwardingHttps()) {
```

Comment:
This might be more readable:
```java
&&  nullToEmpty(request.getHeader(HttpHeaders.X_FORWARDED_PROTO)).equalsIgnoreCase(HTTPS_PROTOCOL)
```

Also I'm pretty sure that the spec allows for the value to be case insensitive (https://tools.ietf.org/html/rfc7239#section-5.4 -> https://tools.ietf.org/html/rfc3986#section-3.1) `Although schemes are case-
   insensitive, the canonical form is lowercase and documents that
   specify schemes must do so with lowercase letters`